### PR TITLE
feat(TDP-3453): preparation dataset sub header

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-3453

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/905/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/905/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/909/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/909/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/910/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/910/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/912/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/912/)

**Comment by charlesnguyen** [comment]
It is missing some styles about some colors but it is ready for review.

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/917/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/917/)

File: dataprep-webapp/src/app/components/export/export.html [hunk 1]
```diff
@@ -12,9 +12,8 @@
   ~
   ~ ============================================================================
   -->
-<div class="export-dropdown btn-group" uib-dropdown>
-    <button id="export-button"
-            type="button"
+<div id="export-button" class="export-dropdown btn-group" uib-dropdown>
+    <button type="button"
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/export/export.html]: `button` is obviously of type button :)

File: dataprep-webapp/src/app/components/playground/header/_playground-header.scss [hunk 2]
```diff
@@ -15,26 +15,76 @@ $playground-preview-icon-size: 20px !default;
...
+	}
+
+	ul > li {
+		display: flex;
+		align-items: center;
+
+		> a {
+			padding: 10px 15px;
+			color: $medium-gray;
+			cursor: pointer;
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/playground/header/_playground-header.scss]: `a` and cursor pointer?

File: dataprep-webapp/src/app/components/playground/header/playground-header-component.js [hunk 3]
```diff
@@ -62,6 +62,9 @@ const PlaygroundHeader = {
 		preparationPicker: '<',
 		preview: '<',
 		isReadonly: '<',
+		preparationName: '<',
+		nameEditionMode: '<',
+		confirmNameEdition: '=',
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/playground/header/playground-header-component.js]: I'm wondering why did you need double binding here?

File: dataprep-webapp/src/app/components/playground/header/playground-header.html [hunk 4]
```diff
@@ -66,47 +49,81 @@
...
+               translate-once-title="LOOKUP_ICON_TOOLTIP">
+                <icon name="'talend-chain'"/>
+            </a>
+        </li>
+        <li id="export-data"
+            ng-if="$ctrl.enableExport"
+            insertion-export>
+            <export id="export-types-dropdown"/>
+        </li>
+        <history-control
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/playground/header/playground-header.html]: It should be wrapped into `<li>` since that component produces a `<ul>`

File: dataprep-webapp/src/app/index-route.js [hunk 5]
```diff
@@ -44,4 +44,22 @@ export default ($stateProvider, $urlRouterProvider) => {
...
+
+export function routeInterceptor($rootScope, $window, $translate, state) {
+	'ngInject';
+
+	$rootScope.$on('$stateChangeSuccess', (event, current) => {
+		if (current.name === PLAYGROUND_PREPARATION_ROUTE &&
+			state.playground.preparation &&
+			state.playground.preparation.name) {
+			// When reload = false
+			$window.document.title = state.playground.preparation.name + ' | ' + $translate.instant('TALEND');
```
Comment (frassinier) [inline dataprep-webapp/src/app/index-route.js]: ES6 string template here

File: dataprep-webapp/src/app/index-route.js [hunk 6]
```diff
@@ -44,4 +44,22 @@ export default ($stateProvider, $urlRouterProvider) => {
...
+		}
+		else {
+			// Reset tab title
+			$window.document.title = $translate.instant('DATA_PREPARATION') + ' | ' + $translate.instant('TALEND');
```
Comment (frassinier) [inline dataprep-webapp/src/app/index-route.js]: ES6 string template here

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 7]
```diff
@@ -131,12 +131,14 @@ export default function PlaygroundService($state, $rootScope, $q, $translate, $t
 		if (preparation) {
 			StateService.showRecipe();
 			ExportService.refreshTypes('preparations', preparation.id);
+			$window.document.title = preparation.name + ' | ' + $translate.instant('TALEND');
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: ES6 string template here

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 8]
```diff
@@ -131,12 +131,14 @@ export default function PlaygroundService($state, $rootScope, $q, $translate, $t
...
 		}
 
 		// dataset specific init
 		else {
 			StateService.setNameEditionMode(true);
 			ExportService.refreshTypes('datasets', dataset.id);
+			$window.document.title = dataset.name + ' | ' + $translate.instant('TALEND');
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: ES6 string template here

File: dataprep-webapp/src/app/components/export/export.html (hunk 1 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/export/export.html]: done

File: dataprep-webapp/src/app/components/playground/header/playground-header-component.js (hunk 3 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/playground/header/playground-header-component.js]: because of the inputs of  <talend-editable-text> directive

File: dataprep-webapp/src/app/components/playground/header/playground-header.html (hunk 4 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/playground/header/playground-header.html]: done

File: dataprep-webapp/src/app/index-route.js (hunk 5 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/index-route.js]: done

File: dataprep-webapp/src/app/index-route.js (hunk 6 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/index-route.js]: done

File: dataprep-webapp/src/app/services/playground/playground-service.js (hunk 8 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: done

File: dataprep-webapp/src/app/components/playground/header/_playground-header.scss (hunk 2 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/playground/header/_playground-header.scss]: since no bootstrap style is applied

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1025/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/1025/)
